src/main/features/claim/routes/pay.ts

```diff
@@ -85,27 +85,29 @@ export default express.Router()
         throw new Error('No amount entered, you cannot pay yet')
       }
 
-      const paymentId = draft.document.claimant.payment.id
+      const paymentRef = draft.document.claimant.payment ? draft.document.claimant.payment.reference : undefined
 
-      if (paymentId) {
+      if (paymentRef) {
         const payClient: PayClient = await getPayClient()
-        const payment: Payment = await payClient.retrieve(user, paymentId)
-        switch (payment.state.status) {
-          case 'success':
-            return res.redirect(Paths.finishPaymentReceiver.evaluateUri({ externalId: draft.document.externalId }))
+        const paymentResponse: PaymentRetrieveResponse = await payClient.retrieve(user, paymentRef)
+        if (paymentResponse !== undefined) {
+          switch (paymentResponse.status) {
+            case 'Success':
```

Comment:
What if the status is different? Shouldn't we have a `default` handler here?

Reply:
In this case we are interested to do quick check whether payment was successful, otherwise we wanna to kick off regular initiate payment logic. Your question made me rename switch and put regular `if` there.